pandas/_libs/tslib.pyx

```diff
@@ -546,17 +546,10 @@ cpdef array_to_datetime(
                     seen_datetime = True
                     iresult[i] = get_datetime64_nanos(val, NPY_FR_ns)
 
-                elif is_integer_object(val) or is_float_object(val):
-                    if require_iso8601:
-                        if is_coerce:
-                            iresult[i] = NPY_NAT
-                            continue
-                        elif is_raise:
-                            raise ValueError(
-                                f"time data \"{val}\" at position {i} doesn't "
-                                f"match format \"{format}\""
-                            )
-                        return values, tz_out
+                elif (
+                    (is_integer_object(val) or is_float_object(val))
+                    and format is None
+                ):
```

Comment:
parsing integers with a format is allowed in the non-ISO8601 path, so let's allow it here too

Note that this simplifies the logic. If the input is an integer/float and a format was specified, it can just be parsed as a string, just like it done here:

https://github.com/pandas-dev/pandas/blob/353d9554746c2797e2cd8587a4342de0e410d124/pandas/_libs/tslibs/strptime.pyx#L143-L170

(in a future PR, I plan to share code between the ISO8601 and non-ISO8601 paths - this'll simplify things and reduce the chances for bugs)